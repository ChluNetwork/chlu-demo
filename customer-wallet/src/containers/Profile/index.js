import React, { Component } from 'react'
import PropTypes from 'prop-types'
// redux
import { connect } from 'react-redux'
import { getProfile } from 'store/modules/data/profile'
// components
import ProfileHeader from './sections/HeaderSection'
import Review from './sections/Review'
// styles
import './styles.css'

class Profile extends Component {

  static propTypes = {
    profile: PropTypes.shape({
      name: PropTypes.string,
      rating: PropTypes.number,
      title: PropTypes.string,
      reviews: PropTypes.array
    }).isRequired,
    getProfile: PropTypes.func.isRequired
  }

  componentDidMount() {
    const { getProfile } = this.props

    getProfile()
  }

  render() {
    const { profile: {
      loading,
      data: { name, rating, title, reviews }
    } } = this.props

    return (
      <div className='page-container profile'>
        {
          loading
            ? 'Loading...'
            : <div>
              <ProfileHeader name={name} rating={rating} title={title} />
              <div>
                {Array.isArray(reviews) && reviews.map((review, idx) => (
                  <Review {...review} key={idx} />
                ))}
              </div>
            </div>

        }
      </div>
    )
  }
}

Profile.propTypes = {
  profile: PropTypes.shape({
    name: PropTypes.string,
    rating: PropTypes.number,
    title: PropTypes.string,
    reviews: PropTypes.array
  }).isRequired,
  getProfile: PropTypes.func.isRequired
}

const mapStateToProps = store => ({
  profile: store.data.profile
})

const mapDispatchToProps = {
  getProfile
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile)
