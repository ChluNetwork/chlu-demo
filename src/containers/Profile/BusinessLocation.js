import React from 'react';
import mapboxgl from 'mapbox-gl';

const STYLE = {
  height: 300,
  marginBottom: 48,
  marginTop: 24
};

mapboxgl.accessToken = 'pk.eyJ1Ijoiam9obndlaXN6IiwiYSI6ImNqa3NheWoyNTQzMHkzcW8zem4waTMyMmkifQ._qzWXiBScWpNONx6BxWvgg';

export default class BusinessLocation extends React.PureComponent {

  render() {
    if (!this.props.location) {
      return null;
    }

    return (
      <div style={STYLE} ref={div => this.container = div} />
    );
  }

  componentDidMount() {
    this.createMapIfNeeded();

    if (this.props.location) {
      this.addMapMarkerFromCurrentLocation();
    }
  }

  componentDidUpdate(prevProps) {
    if (!this.props.location && this.map) {
      this.map.remove();
    } else if (this.props.location !== prevProps.location && this.map) {
      // Location changing, need new coordinates for map.
      this.addMapMarkerFromCurrentLocation();
    } else {
      this.createMapIfNeeded();
      this.addMapMarkerFromCurrentLocation();
    }
  }

  componentWillUnmount() {
    if (this.map) {
      this.map.remove();
    }
  }

  createMapIfNeeded() {
    if (this.container && !this.map) {
      this.map = new mapboxgl.Map({
        container: this.container,
        style: 'mapbox://styles/mapbox/streets-v9',
      });
    }
  }

  addMapMarkerFromCurrentLocation() {
    let { latitude, longitude, bounds } = this.props.location;

    if (this.map) {
      let markerElem = document.createElement("div");

      markerElem.style.width = "24px";
      markerElem.style.height = "24px";
      markerElem.style.borderRadius = "24px";
      markerElem.style.backgroundColor = "red";
      markerElem.style.border = "2px solid white";

      if (this.locationMarker) {
        this.locationMarker.remove();
      }

      this.locationMarker = new mapboxgl.Marker(markerElem).setLngLat([latitude, longitude]).addTo(this.map);
      this.map.fitBounds(bounds);
    }
  }
}
