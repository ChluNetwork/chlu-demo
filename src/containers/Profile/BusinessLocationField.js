import React from "react";

import CustomInput from 'components/MaterialDashboardPro/CustomInput';
import ExploreIcon from '@material-ui/icons/Explore';
import { List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core'
import { autocomplete } from 'helpers/geocode';
import { debounce } from 'helpers/debounce';

export default class BusinessLocationField extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.updateSuggestions = debounce(this.updateSuggestions, 800); // TODO: this should be done with a decorator instead.

    this.state = {
      suggestions: []
    };
  }

  handleInputChange = (e) => {
    if (this.props.onChange) {
      this.props.onChange(e.target.value);
    }

    this.updateSuggestions(e.target.value);
  }

  handleInputKeyDown = (e) => {
    if (e.which === 13 && this.state.suggestions.length > 0) {
      if (this.props.onChange) {
        this.props.onChange(this.state.suggestions[0]);
      }

      this.setState({
        suggestions: []
      });
    }
  }

  handleSuggestionClick = (value) => {
    this.setState({
      suggestions: []
    });

    if (this.props.onChange) {
      this.props.onChange(value);
    }
  }

  updateSuggestions = async (value) => {
    let suggestions = await autocomplete(value);

    this.setState({
      suggestions: suggestions
    });
  }

  render() {
    return (
      <div>
        <CustomInput
          labelText='Where is your business located?'
          id='location'
          formControlProps={{ fullWidth: true }}
          inputProps={{ onChange: this.handleInputChange, value: this.props.value, onKeyDown: this.handleInputKeyDown }}
        />
        {this.renderSuggestions()}
      </div>
    )
  }

  renderSuggestions() {
    if (this.state.suggestions.length > 0) {
      return (
        <div>
          <List>
            {this.state.suggestions.map((suggestion, i) => this.renderSuggestionItem(suggestion, i))}
          </List>
        </div>
      )
    }
  }

  renderSuggestionItem(suggestion, i) {
    return (
      <ListItem key={i} button onClick={() => this.handleSuggestionClick(suggestion)}>
        <ListItemIcon>
          <ExploreIcon />
        </ListItemIcon>
        <ListItemText primary={suggestion} secondary={i === 0 ? "Best match, press ENTER to accept" : undefined} />
      </ListItem>
    )
  }
}
