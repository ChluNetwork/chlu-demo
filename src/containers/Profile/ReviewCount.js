import React, { Component } from 'react'
import { withStyles } from '@material-ui/core'
import CheckIcon from '@material-ui/icons/CheckCircle'

const styles = theme => ({
  reviewCount: {
    color: 'rgb(15,155,15)',
    fontSize: 14,
    fontWeight: 500,
    display: 'inline-block',
    paddingLeft: theme.spacing.unit * 2
  },
  reviewCountUnverified: {
    color: 'rgba(0,0,0,0.5)',
    fontSize: 14,
    display: 'inline-block',
    paddingLeft: theme.spacing.unit * 2
  },
  reviewCountVerifiedIcon: {
    height: 16,
    width: 16,
    margin: 0,
    marginTop: -2,
    verticalAlign: 'middle'
  },
  reviewCountTotal: {
    color: 'rgba(0,0,0,0.5)',
    fontSize: 14,
    fontWeight: 100
  }
});

class ReviewCount extends Component {
  render() {
    const { classes, reviews } = this.props
    const count = reviews ? reviews.length : 0
    const verifiableCount = reviews ? reviews.filter(review => review.verifiable).length : 0

    if (count > 0) {
      if (verifiableCount > 0) {
        return (
          <span className={classes.reviewCount} ref={span => this.verifiedReviewCounterElem = span}>
            <CheckIcon className={classes.reviewCountVerifiedIcon} /> {this.verifiableCountText(verifiableCount)} {this.totalCountText(count, verifiableCount)}
          </span>
        )
      } else {
        return (
          <span className={classes.reviewCountUnverified}>
            {this.countText(count)}
          </span>
        )
      }
    } else {
      return (
        <span className={classes.reviewCountUnverified}>
          No reviews yet
        </span>
      )
    }
  }

  componentDidMount() {
    if (this.verifiedReviewCounterElem && this.verifiedReviewCounterElem.animate) {
      this.verifiedReviewCounterElem.animate(
        [
          { transform: 'scale(0)' },
          { transform: 'scale(1.3)' },
          { transform: 'scale(1)' }
        ],
        {
          duration: 900,
          easing: 'cubic-bezier(0.6,0,0,1)',
          delay: 600,
          fill: 'backwards'
        }
      )
    }
  }

  verifiableCountText(verifiableCount) {
    const { hasMoreReviews } = this.props

    if (hasMoreReviews) {
      if (verifiableCount === 1) {
        return 'More than 1 verifiable review'
      } else {
        return `More than ${verifiableCount} verifiable reviews`
      }
    } else {
      if (verifiableCount === 1) {
        return '1 verifiable review'
      } else {
        return `${verifiableCount} verifiable reviews`
      }
    }
  }

  countText(count) {
    const { hasMoreReviews } = this.props

    if (hasMoreReviews) {
      if (count === 1) {
        return 'More than 1 review'
      } else {
        return `More than ${count} reviews`
      }
    } else {
      if (count === 1) {
        return '1 review'
      } else {
        return `${count} reviews`
      }
    }
  }

  totalCountText(count, verifiedCount) {
    const { classes } = this.props

    if (count > verifiedCount) {
      if (this.props.hasMoreReviews) {
        return (
          <span className={classes.reviewCountTotal}>
            (more than {count} total)
          </span>
        )
      } else {
        return (
          <span className={classes.reviewCountTotal}>
            ({count} total)
          </span>
        )
      }
    }
  }
}

export default withStyles(styles)(ReviewCount)
