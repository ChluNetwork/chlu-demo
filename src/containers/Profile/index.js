import React, { Component } from 'react'
// Components
import { withStyles, Card, CardContent, Tabs, Tab, LinearProgress } from '@material-ui/core'
import Payment from 'containers/Payment'
import Reviews from 'components/Reviews'
// icons
import ProfileIcon from '@material-ui/icons/AccountCircle'
import ReviewsIcon from '@material-ui/icons/Star'
import PaymentIcon from '@material-ui/icons/AccountBalanceWallet'
import Profile from './Profile';
// helpers
import { get } from 'lodash'
import profileProvider from 'helpers/profileProvider';
// redux
import { compose } from 'recompose';
import { push } from 'react-router-redux'
import { connect } from 'react-redux'
import { readReputation } from 'store/modules/data/reputation'

const styles = {
  card: {
    margin: '30px'
  }
}
class ProfileContainer extends Component {

  constructor(props) {
    super(props)
    this.state = {
      loading: true,
      profile: {}
    }
  }

  updateProfileAndReviews() {
    const didId = this.props.match.params.id
    if (didId) {
      profileProvider.getProfile(didId).then(profile => this.setState({ profile: profile || {}, loading: false }));
      this.props.readReputation(didId)
    }
  }

  componentDidMount() {
    this.updateProfileAndReviews()
  }

  componentDidUpdate(prevProps) {
    const didId = this.props.match.params.id;
    if (didId !== prevProps.match.params.id) {
      this.setState({ loading: true }, () => this.updateProfileAndReviews())
    }
  }

  handleTabChange = (event, value) => {
    const id = this.props.match.params.id
    const urls = [
      `/profile/${id}`,
      `/profile/${id}/reputation`,
      `/profile/${id}/pay`,
    ]
    if (urls[value]) {
      this.props.push(urls[value])
    }
  };

  render() {
    const {
      match,
      classes,
      page = 'profile',
      reviews,
      reviewsLoading,
      reviewsDidId,
      wallet
    } = this.props
    const { profile, loading: loadingProfile } = this.state
    const didId = match.params.id;
    const tabIndex = [
      'profile',
      'reputation',
      'payment'
    ].indexOf(page)
    const myDid = get(wallet, 'did.publicDidDocument.id', null)
    const showProfile = !loadingProfile && profile
    const showPayment = !loadingProfile && myDid && didId !== myDid 
    const showReviews = !reviewsLoading && reviewsDidId === didId
    const loading = !(showProfile && showReviews && showPayment)

    return <Card className={classes.card}>
      <Tabs
        value={tabIndex}
        onChange={this.handleTabChange}
        fullWidth
        indicatorColor='secondary'
        textColor='secondary'
        centered
      >
        <Tab icon={<ProfileIcon className={classes.rightIcon}/>} label='Profile' />
        <Tab icon={<ReviewsIcon className={classes.rightIcon}/>} label='Reviews' />
        <Tab icon={<PaymentIcon className={classes.rightIcon}/>} label='Payment' />
      </Tabs>
      <CardContent>
        { loading && <LinearProgress/>}
        { tabIndex === 0 && !loading && <Profile profile={profile} /> }
        { tabIndex === 1 && !loading && <Reviews reviews={reviews} /> }
        { tabIndex === 2 && !loading && <Payment profile={profile} didId={didId} /> }
      </CardContent>
    </Card>
  }
}

const mapStateToProps = store => {
  return {
    wallet: store.data.wallet,
    reviews: store.data.reputation.reviews,
    reviewsDidId: store.data.reputation.didId,
    reviewsLoading: store.data.reputation.loading
  };
};

const mapDispatchToProps = {
  readReputation,
  push
}

export default compose(
  withStyles(styles),
  connect(mapStateToProps, mapDispatchToProps)
)(ProfileContainer)