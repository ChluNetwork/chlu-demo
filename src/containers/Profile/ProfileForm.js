import React, { Component } from 'react'
// components
import {
  withStyles,
  InputAdornment,
  Grid,
  FormControl,
  InputLabel,
  MenuItem,
  Select
} from '@material-ui/core'
import ProfileImageUpload from 'components/ProfileImageUpload'
import CustomInput from 'components/Form/CustomInput';
// icons
import Face from '@material-ui/icons/Face'
import Email from '@material-ui/icons/Email'
import Phone from '@material-ui/icons/Phone'
import Web from '@material-ui/icons/Web'
import BusinessLocationField from './BusinessLocationField';
// style
import customSelectStyle from 'styles/material-dashboard-pro-react/customSelectStyle.jsx';
import customCheckboxRadioSwitch from 'styles/material-dashboard-pro-react/customCheckboxRadioSwitch.jsx';
// data
import { businessTypes } from 'store/modules/ui/profile';
// redux
import { compose } from 'recompose'
import { reduxForm, Field } from 'redux-form'
// validation
import { validateProfile } from 'chlu-marketplace-js/src/profile'

const ProfileImageUploadForm = props => {
  const { input: { value, onChange } } = props
  return <ProfileImageUpload
    imageDataUrl={value}
    onChoosePicture={onChange}
  />
}

const SelectForm = props => {
  const { input: { value, onChange }, helpText, options, classes } = props
  const valueIndex = options.indexOf(value)
  return <FormControl fullWidth className={classes.selectFormControl}>
      <InputLabel htmlFor='simple-select' className={classes.selectLabel}>
        {helpText}
      </InputLabel>
      <Select
        MenuProps={{ className: classes.selectMenu }}
        classes={{ select: classes.select }}
        value={valueIndex >= 0 ? valueIndex : 0}
        onChange={e => onChange(options[e.target.value])}
      >
        {options.map((v, i) => (
          <MenuItem
            key={i}
            value={i}
            disabled={!i}
            classes={{ root: classes.selectMenuItem, selected: i ? classes.selectMenuItemSelected : undefined }}
          >{v}</MenuItem>
        ))}
      </Select>
    </FormControl>
}

const BusinessLocationFieldForm = props => {
  const { input: { value, onChange }, ...others } = props
  return <BusinessLocationField value={value} onChange={onChange} {...others} />
}

class ProfileForm extends Component {

  render() {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        {this.renderUser()}
        {this.renderBusiness()}
      </form>
    )
  }

  renderUser() {
    const { classes, userType } = this.props
    if (userType !== 'individual') return undefined
    return (
      <Grid container justify='center' spacing={16}>
        <Grid item xs={12} sm={12} md={12}>
          <Field
            name='avatarDataUrl'
            component={ProfileImageUploadForm}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='email'
            type='email'
            component={CustomInput}
            labelText={<span>Email</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Email className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='username'
            component={CustomInput}
            labelText={<span>User Name</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Face className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='firstname'
            component={CustomInput}
            labelText={<span>First Name</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Face className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='lastname'
            component={CustomInput}
            labelText={<span>Last Name</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              onChange: event => this.change(event.target.value, 'lastname', 'length', 3),
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Face className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>
      </Grid>
    )
  }

  renderBusiness() {
    const { classes, userType } = this.props
    if (userType !== 'business') return undefined
    return (
      <Grid container justify='center' spacing={16}>
        <Grid item xs={12} sm={12} md={12}>
          <Field
            name='avatarDataUrl'
            component={ProfileImageUploadForm}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='businessname'
            component={CustomInput}
            labelText={<span>Business Name</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Face className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={5}>
          <Field
            name='businesstype'
            component={SelectForm}
            labelText={<span>Business Type</span>}
            options={businessTypes}
            classes={classes}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={3}>
          <Field
            name='email'
            component={CustomInput}
            labelText={<span>Email</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Email className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={4}>
          <Field
            name='website'
            component={CustomInput}
            labelText={<span>Website</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Web className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={3}>
          <Field
            name='phone'
            component={CustomInput}
            labelText={<span>Phone</span>}
            formControlProps={{ fullWidth: true }}
            inputProps={{
              endAdornment: (
                <InputAdornment position='end' className={classes.inputAdornment}>
                  <Phone className={classes.inputAdornmentIcon} />
                </InputAdornment>
              )
            }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={10}>
          <Field
            name='description'
            component={CustomInput}
            labelText='A brief description of your business.'
            formControlProps={{ fullWidth: true }}
          />
        </Grid>

        <Grid item xs={12} sm={12} md={10}>
          <Field
            name='businesslocation'
            component={BusinessLocationFieldForm}
            classes={classes}
          />
        </Grid>
      </Grid>
    )
  }
}

const style = {
  infoText: {
    fontWeight: '300',
    margin: '10px 0 30px',
    textAlign: 'center'
  },
  inputAdornmentIcon: {
    color: '#555'
  },
  choice: {
    textAlign: 'center',
    cursor: 'pointer',
    marginTop: '20px'
  },
  description: {
    textAlign: 'center'
  },
  ...customSelectStyle,
  ...customCheckboxRadioSwitch
};

const validate = (profile, props) => {
  const preparedProfile = {
    type: props.userType,
    ...profile
  }
  // Here we use backend code to perform frontend validation. How cool is that?
  return validateProfile(preparedProfile) || {}
}

export default compose(
  withStyles(style),
  reduxForm({
    form: 'profile',
    validate
  })
)(ProfileForm)