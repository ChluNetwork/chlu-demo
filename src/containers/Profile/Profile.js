import React, { Component } from 'react'

// components
import { CardHeader, Grid, List, ListItem, ListItemIcon, ListItemText, Paper } from '@material-ui/core'
import { Avatar, withStyles } from '@material-ui/core'
import PictureUpload from 'components/MaterialDashboardPro/PictureUpload'

// icons
import EmailIcon from '@material-ui/icons/Email'
import PhoneIcon from '@material-ui/icons/Phone'
import WebIcon from '@material-ui/icons/Web'
import PlaceIcon from '@material-ui/icons/Place'
import ProfileIcon from '@material-ui/icons/AccountCircle'
import BusinessLocation from './BusinessLocation'
import ReviewCount from './ReviewCount';

const styles = theme => ({
  profileRow: {
    paddingTop: '7%'
  },
  profileContainer: {
    alignItems: 'flex-start'
  },
  fullname: {
    color: 'rgba(0,0,0,0.8)',
    fontSize: 36,
    marginBottom: theme.spacing.unit
  },
  username: {
    color: 'rgba(0,0,0,0.5)',
    fontSize: 14,
    paddingTop: theme.spacing.unit
  },
  description: {
    borderTop: `1px solid rgba(0,0,0,0.1)`,
    color: 'rgba(0,0,0,0.6)',
    fontSize: 18,
    fontStyle: 'italic',
    paddingTop: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit
  },
  profilePic: {
    pointerEvents: 'none'
  },
  map: {
    position: 'relative',
    marginTop: '5%'
  },
  mapShadowOverlay: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
    pointerEvents: 'none',
    zIndex: 999,
    boxShadow: '0 1px 10px rgba(0,0,0,0.3) inset'
  },
  contactCard: {
    marginTop: theme.spacing.unit * 4,
    marginBottom: theme.spacing.unit * 2
  }
});

class Profile extends Component {
  render () {
    const { loading } = this.props

    return(
      <div>
        { loading && <CardHeader
          avatar={<Avatar><ProfileIcon/></Avatar>}
          title='Profile'
          subheader='Loading...'
        /> }
        {!loading && this.renderUser()}
        {!loading && this.renderBusiness()}
      </div>
    )
  }

  renderUser() {
    const { classes, profile, reviewsCount } = this.props;

    if (profile.businessname || profile.location) return undefined;

    return (
      <div className={classes.profileRow}>
        <Grid container justify='center' alignItems='flex-start' spacing={16}>
          <Grid item xs={12} sm={12} md={2} className={classes.profilePic}>
            <PictureUpload />
          </Grid>

          <Grid item xs={12} sm={12} md={8}>
            <div className={classes.fullname}>
              {profile.firstname} {profile.lastname}
            </div>

            <div className={classes.username}>
              @{profile.username} <ReviewCount count={reviewsCount} />
            </div>

            {this.renderContactOptions()}
          </Grid>
        </Grid>
      </div>
    )
  }

  renderBusiness() {
    const { profile, classes, reviewsCount } = this.props

    if (!profile.businessname && !profile.location) return undefined;

    return (
      <div className={classes.profileRow}>
        <Grid container justify='center' alignItems='flex-start' spacing={16}>
          <Grid item xs={12} sm={12} md={2} className={classes.profilePic}>
            <PictureUpload />
          </Grid>

          <Grid item xs={12} sm={12} md={8}>
            <div className={classes.fullname}>
              {profile.businessname}
            </div>

            <div className={classes.username}>
              {profile.businesstype} <ReviewCount count={reviewsCount} />
            </div>

            <div className={classes.description}>
              {profile.businessdescription}
            </div>

            {this.renderContactOptions()}
          </Grid>

          <Grid item xs={12} sm={12} md={12}>
            <div className={classes.map}>
              <BusinessLocation location={profile.businesslocationgeo} />
              <div className={classes.mapShadowOverlay} />
            </div>
          </Grid>
        </Grid>
      </div>
    )
  }

  renderContactOptions() {
    return (
      <div className={this.props.classes.contactCard}>
        <List>
          {this.renderEmail()}
          {this.renderPhone()}
          {this.renderLocation()}
          {this.renderUrl()}
        </List>
      </div>
    )
  }

  renderEmail() {
    const { profile } = this.props

    if (profile.email) {
      return (
        <ListItem>
          <ListItemIcon><EmailIcon /></ListItemIcon>
          <ListItemText primary={profile.email} />
        </ListItem>
      )
    }
  }

  renderPhone() {
    const { profile } = this.props

    if (profile.phone) {
      return (
        <ListItem>
          <ListItemIcon><PhoneIcon /></ListItemIcon>
          <ListItemText primary={profile.phone} />
        </ListItem>
      )
    }
  }

  renderUrl() {
    const { profile } = this.props

    if (profile.website) {
      return (
        <ListItem>
          <ListItemIcon><WebIcon /></ListItemIcon>
          <ListItemText primary={profile.website} />
        </ListItem>
      )
    }
  }

  renderLocation() {
    const { profile } = this.props

    if (profile.businesslocation) {
      return (
        <ListItem>
          <ListItemIcon><PlaceIcon /></ListItemIcon>
          <ListItemText primary={profile.businesslocation} />
        </ListItem>
      )
    }
  }
}

export default withStyles(styles)(Profile)
