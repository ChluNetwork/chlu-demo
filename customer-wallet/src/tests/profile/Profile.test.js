import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Profile from 'containers/Profile'

const mockStore = configureStore([])
const store = mockStore({
  data: {
    profile: {
      loading: false,
      error: null,
      data: {}
    }
  }
})

describe('<Profile />', () => {

  it('should render correctly profile container', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Profile />
      </Provider>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
