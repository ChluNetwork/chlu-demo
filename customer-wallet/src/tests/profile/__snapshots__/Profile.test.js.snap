// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render correctly profile container 1`] = `<Connect(Profile) />`;
